packages/presentation/lib/screen/login/bloc/login_validate/validation.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:data/core/error_const.dart';
+
+String validatePassword(String formPassword, String exceptionPassword) {
+  if (formPassword.isEmpty) return "password is requried.";
+  String pattern =
+      r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$';
+  RegExp reg = RegExp(pattern);
+  if (!reg.hasMatch(formPassword)) {
+    return ErrorTextField.passwordInvalid;
+  }
+  return exceptionPassword;
+}
+
+String validateEmail(String formEmail, String exceptionEmail) {
+  if (formEmail.isEmpty) return "E-mail is requried.";
```

Comment:
keep '"E-mail is requried." and the same params in constants file;

Reply:
I fixed this in login/networking